tests/utils.py

```diff
@@ -0,0 +1,128 @@
+import itertools
```

Comment:
Please add a module level docstring followed by the stand `__future__` imports. We should document the public functions and classes also.